erpnext/payroll/doctype/gratuity/gratuity.py

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe import _, bold
+from frappe.model.document import Document
+from frappe.utils import flt, get_datetime, get_link_to_form
+from math import floor
+
+from frappe.utils import get_datetime
+class Gratuity(Document):
+	def validate(self):
+		data = calculate_work_experience_and_amount(self.employee, self.gratuity_rule)
+		self.current_work_experience = data["current_work_experience"]
+		self.amount = data["amount"]
+
+	def before_submit(self):
+		self.status = "Unpaid"
+		if self.pay_via_salary_slip:
+			self.status = "Paid"
```

Comment:
Why? It should be paid only if it is actually paid via salary. Check if we can back update the status.

Reply:
done